src/main/psxsdk/libgpu/sys.c

```diff
@@ -1,61 +1,259 @@
 INCLUDE_ASM("asm/us/main/nonmatchings/psxsdk/libgpu/sys", ResetGraph);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/psxsdk/libgpu/sys", SetGraphReverse);
+int SetGraphReverse(int mode) {
+    unsigned int var_a0;
+    s32 temp_a0;
+    s32 var_v0;
+    int old = D_8002C270;
+
+    if (D_8002C268 >= 2) {
+        GPU_printf("SetGraphReverse(%d)...\n", mode);
+    }
+
+    D_8002C270 = mode;
+    D_8002C260->ctl(
+        D_8002C260->getctl(8) | (D_8002C270 ? 0x08000080 : 0x08000000));
+    if (D_8002C26C == 2) {
+        D_8002C260->ctl(D_8002C270 ? 0x20000501 : 0x20000504);
+    }
+    return old;
+}
+
+int SetGraphDebug(int level) {
+    int old = D_8002C268;
+
+    D_8002C268 = level;
+    if (level) {
+        GPU_printf("SetGraphDebug:level:%d,type:%d reverse:%d\n", level,
+                   D_8002C26C, D_8002C270);
+    }
+
+    return old;
+}
+
+int SetGraphQueue(int mode) {
+    int old = D_8002C274;
+
+    if (D_8002C268 >= 2) {
+        GPU_printf("SetGraphQueue(%d)...\n", mode);
+    }
+    if (mode != D_8002C274) {
+        D_8002C260->reset(1);
+        D_8002C274 = mode;
+    }
+
+    return old;
+}
+
+int GetGraphType(void) { return D_8002C26C; }
+
+int GetGraphDebug(void) { return D_8002C268; }
+
+u_long DrawSyncCallback(void (*func)()) {
+    void (*old)();
+
+    if (D_8002C268 >= 2) {
+        GPU_printf("DrawSyncCallback(%08x)...\n", func);
+    }
+    old = D_8002C278;
+    D_8002C278 = func;
+    return old;
+}
+
+void SetDispMask(int mask) {
+    if (D_8002C268 >= 2) {
+        GPU_printf("SetDispMask(%d)...\n", mask);
+    }
+    D_8002C260->ctl(mask ? 0x03000000 : 0x03000001);
+}
+
+s32 DrawSync(s32 mode) {
+    if (D_8002C268 >= 2) {
+        GPU_printf("DrawSync(%d)...\n", mode);
+    }
+    D_8002C260->sync(mode);
+}
+
+void checkRECT(const char* log, RECT* r) {
+    switch (D_8002C268) {
+    case 1:
+        if (r->w > 0x400 || r->w + r->x > 0x400 || r->y > 0x400 ||
+            r->y + r->h > 0x400 || r->w <= 0 || r->x < 0 || r->y < 0 ||
+            r->h <= 0) {
+            GPU_printf("%s:bad RECT", log);
+            GPU_printf("(%d,%d)-(%d,%d)\n", r->x, r->y, r->w, r->h);
+        }
+        break;
+    case 2:
+        GPU_printf("%s:", log);
+        GPU_printf("(%d,%d)-(%d,%d)\n", r->x, r->y, r->w, r->h);
+        break;
+    }
+}
+
+int ClearImage(RECT* rect, u8 r, u8 g, u8 b) {
+    checkRECT("ClearImage", rect);
+    return D_8002C260->addque2(
+        D_8002C260->clr, rect, sizeof(RECT), (b << 0x10) | (g << 8) | r);
+}
+
+int LoadImage(RECT* rect, u_long* p) {
+    checkRECT("LoadImage", rect);
+    return D_8002C260->addque2(D_8002C260->dws, rect, sizeof(RECT), p);
+}
+
+int StoreImage(RECT* rect, u_long* p) {
+    checkRECT("StoreImage", rect);
+    return D_8002C260->addque2(D_8002C260->drs, rect, sizeof(RECT), p);
+}
+
+int MoveImage(RECT* rect, int x, int y) {
+    u32 param[5];
+
+    checkRECT("MoveImage", rect);
+    param[0] = 0x04FFFFFF;
+    param[1] = 0x80000000;
+    param[2] = *(u32*)&rect->x;
```

Comment:
Does `rect->x << 16 | rect->y` or the swapped version of that match? Wondering if this is an optimization. I looked through the SDK docs and didn't see any mention of an alternate meaning for RECT->x